sdk/typespec/typespec_client_core/src/macros.rs

```diff
@@ -1,6 +1,100 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+/// The following macro invocation:
+/// ```
+/// # #[macro_use] extern crate typespec_client_core;
+/// create_enum!(Words, (Chicken, "Chicken"), (White, "White"), (Yellow, "Yellow"));
+/// ```
+/// Turns into a struct where each variant can be turned into and construct from the corresponding string.
+#[macro_export]
+macro_rules! create_enum {
```

Comment:
nit: `create_enum` feels like a very generic name to me. I'd prefer something a little more descriptive like `create_string_enum`.

I know you're just moving an existing macro though, so I'm totally fine with you leaving this as-is or deferring that change to a separate PR.

Reply:
I plan to replace this with a derive macro at some point anyway. This is more so for creating "normal" and "extensible" enums for service methods, to which I've provided guidelines for Rust and C# as well (back in the day).